typing/ctype.ml

```diff
@@ -1578,6 +1577,7 @@ let check_abbrev_env env =
   if not (Env.same_type_declarations env !previous_env) then begin
     (* prerr_endline "cleanup expansion cache"; *)
     cleanup_abbrev ();
+    simple_abbrevs := Mnil;
```

Comment:
It is orthogonal indeed (i.e., it does not apply to this specific case), but I have been always wondering why this caused no bug, and could never find something else to put it in. Since we are fixing bugs about cleaning abbrevs, let's do it properly.